pallets/loans/src/functions.rs

```diff
@@ -754,7 +752,7 @@ impl<T: Config> Pallet<T> {
 					.interest_rate_per_sec
 					.checked_add(&write_off_penalty_rate)
 					.ok_or(ArithmeticError::Overflow)?;
-				T::InterestAccrual::reference_rate(interest_rate_with_penalty);
+				T::InterestAccrual::reference_rate(interest_rate_with_penalty)?;
```

Comment:
Adding write-off groups with penalty rates of e.g. `0.00001` is not checked, right? So this would break or invariant of only having four decimal interest rates in the system.